Simplenote/src/main/java/com/automattic/simplenote/widgets/SimplenoteEditText.java

```diff
@@ -92,6 +103,7 @@ public void run() {
                             && selectionStart <= editable.length()
                             && selectionEnd <= editable.length()) {
```

Comment:
```suggestion
                            && selectionEnd <= editable.length() && hasFocus()) {
```

During my testing, these conditions are always `true`, which means the `setCursorVisible(true);` statement is always called.  That results in the cursor appearing randomly when it shouldn't.  Follow the steps below to reproduce the random cursor appearance.

1. Open any note with checklist.
2. Notice cursor is not shown.
2. Tap only checkbox to toggle checked/unchecked.
3. Repeat Step 2 indefinitely.
4. Notice cursor is shown/hidden before first character in note randomly.

The random cursor appearance can reproduced after selecting anywhere in the note as well.  Follow the steps below to reproduce that case.

1. Open any note with checklist.
2. Tap anywhere in the note other than a checkbox.
3. Notice cursor and soft keyboard are shown.
4. Tap back button to dismiss soft keyboard.
5. Notice cursor is not shown.
6. Tap only checkbox to toggle checked/unchecked.
7. Repeat Step 6 indefinitely.
8. Notice cursor is shown/hidden before first character in note randomly.

In order to stop the random cursor appearance, we could add the `&& hasFocus()` condition to this `if` statement.  Consequently, that would also cause the `setCursorVisible(true);` statement to be called only when the view has focus.  That means if the back button is pressed or a checkbox is toggled, the cursor will no longer be shown.  To fix that, we can override `onFocusChanged` in `SimplenoteEditText` and call `setCursorVisible(true);` if the view has focus like this.

```java
@Override
protected void onFocusChanged(boolean focused, int direction, Rect previouslyFocusedRect) {
    if (focused) {
        setCursorVisible(true);
    }

    super.onFocusChanged(focused, direction, previouslyFocusedRect);
}
```

Let me know if you see the same random cursor behavior described above and what you think about the fix.

Reply:
Hey @theck13, you're right, I'm seeing the same behavior.
I've applied the changes you suggest and it looks good after some testing!